src/test/util_tests.cpp

```diff
@@ -49,24 +49,27 @@ BOOST_FIXTURE_TEST_SUITE(util_tests, BasicTestingSetup)
 
 BOOST_AUTO_TEST_CASE(util_datadir)
 {
-    ClearDatadirCache();
-    const fs::path dd_norm = GetDataDir();
+    ArgsManager argsManager;
```

Comment:
In commit "Modify "util_datadir" unit test to not use gArgs." (ec7769315f5ed62e6145492977226925e122292c)

It seems unnecessary to add another argsmanager for this test, with 3 side-by-argsmanagers:

1. gArgs
2. BasicTestingSetup::m_args_manager added in d4f86a3f608c98365b537d78d7ef1693463d7b9b
3. This local argsManager variable.

Maybe it could just use the basictesting member. If not though, it'd be helpful to have a comment explaining its purpose.

Also see naming in https://github.com/bitcoin/bitcoin/blob/master/doc/developer-notes.md but camelCase should not be used for variable names. If there needs to be a variable would suggest just calling it `args` which is convention for other ArgsManager instances.

Reply:
The main reason I did it this way was/is that I persuades me that it does what one expects and there is no possible confusion. When I put `gArgs` (or `BasicTestingSetup::m_args_manager`) there, it forces to me wonder: How is `gArgs` actually initialized and where it is? Can some later refactoring affect the test silently? 

Maybe I see it this way because I don't read Bitcoin Core source code for years as other people here. Anyway, I'm open to suggestions, if you still find it better to change the code, I can do it.

_______
Note: An alternative would be to move the test somewhere where there is no `BasicTestingSetup` and no `gArgs` but I'm not sure if there are such tests.

Reply:
Added comment, thanks.